macros/src/derive_fn.rs

```diff
@@ -77,11 +77,24 @@ pub fn derive_fn(
                         #ident
                     ).unwrap()
                 };
-                (spec, arg, call)
+                let pat_ty = &pat_type.ty;
+                let invoke_arg = FnArg::Typed(PatType {
+                    attrs: vec![],
+                    pat: Box::new(Pat::Ident(PatIdent {
+                        ident: ident.clone(),
+                        attrs: vec![],
+                        by_ref: None,
+                        mutability: None,
+                        subpat: None,
+                    })),
+                    colon_token: Colon::default(),
+                    ty: Box::new(Type::Verbatim(quote! { super::#pat_ty })),
+                });
+                (spec, arg, call, invoke_arg, quote! { #ident })
```

Comment:
It helps with readability in the long tuple (too long!) if we name the args here.
```suggestion
                let invoke_call = quote! { #ident };
                (spec, arg, call, invoke_arg, invoke_call)
```